Snippets/Snippets_6/UpgradeGuides/5to6/CustomErrorHandlingBehaviour.cs

```diff
@@ -0,0 +1,59 @@
+namespace Snippets6.UpgradeGuides._5to6
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Pipeline;
+
+    #region 5to6-customerrorhandlingbehaviour
+    class CustomErrorHandlingBehaviour : Behavior<ITransportReceiveContext>
+    {
+        public override async Task Invoke(ITransportReceiveContext context, Func<Task> next)
+        {
+            try
+            {
+                await next().ConfigureAwait(false);
+            }
+            catch (MessageDeserializationException)
+            {
+                Console.WriteLine("CustomFaultManager - MessageDeserializationException");
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("CustomFaultManager - ProcessingAlwaysFailsForMessage");
+                // do not rethrow the message if you want to mark the message as processed.
+                // rethrow to move the message to the error queue.
+                throw;
+                // if you want to rollback the receive operation instead of mark as processed:
+                //context.AbortReceiveOperation();
```

Comment:
Fair call
On Fri, 11 Mar 2016 at 6:38 PM, Mike Minutillo <notifications@github.com>
wrote:

> In Snippets/Snippets_6/UpgradeGuides/5to6/CustomErrorHandlingBehaviour.cs
> <https://github.com/Particular/docs.particular.net/pull/1300#discussion_r55798136>
> :
>
> > +            try
> > +            {
> > +                await next().ConfigureAwait(false);
> > +            }
> > +            catch (MessageDeserializationExceptio
>
> n)
> > +            {
> > +                Console.WriteLine("CustomFaultManager - MessageDeserializationException");
> > +            }
> > +            catch (Exception)
> > +            {
> > +                Console.WriteLine("CustomFaultManager - ProcessingAlwaysFailsForMessage");
> > +                // do not rethrow the message if you want to mark the message as processed.
> > +                // rethrow to move the message to the error queue.
> > +                throw;
> > +                // if you want to rollback the receive operation instead of mark as processed:
> > +                //context.AbortReceiveOperation();
>
> I'd break this into 3 different snippets. One that throws, one that
> doesn't, and one that aborts. Then explain the difference in the text
> between. You could then ditch the Console.WriteLine statements as well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/docs.particular.net/pull/1300/files#r55798136>
> .
>